gpkit/form_program.py

```diff
@@ -0,0 +1,59 @@
+import numpy as np
+
+from .nomials import Signomial
+from .geometric_program import GeometricProgram
+from .signomial_program import SignomialProgram
+from .nomial_data import sort_and_simplify
+
+from .substitution import substitution
+from .small_scripts import mag
+
```

Comment:
Overall comment: I'm a bit critical of this function and whether it needs to exist -- detailed comments below.